accessibility/alt-text.md

```diff
@@ -47,6 +49,26 @@ No. If an image has no meaning, or if it's providing information that's redundan
 
 An image MUST always have an alt _attribute_, even if its value is blank (`alt=""`). Your HTML isn't valid without one. 
 
+## How long should the alt text be? 
+
+The length any given alt text _should_ be depends on how long the alt text _needs_ to be. 
+
+Despite what you may have read or heard in the past, [there is no hard character limit imposed by browsers or screen readers](https://yatil.net/blog/there-is-no-character-limit-for-alt-text). This doesn't mean you should write a novel every time. A simple logo will need a lot less alt text than a chart or a graph as the logo image itself carries less information. 
+
+You MUST keep alt text in HTML as succinct as you reasonably can while still expressing all the information in the image. To understand why, you need to understand how screen readers parse attributes like `alt`. 
+
+### How do screen readers render alt text? 
+
+Screen readers look at all of the HTML markup on a page, then announce the contents to the user in different ways depending on what kind of element it's describing. For example a link might be announced as something like "Link: my cool website address". A navigation menu might be announced like "Navigation: breadcrumb". These are examples of HTML element nodes. Screen reader users can command their screen reader to move around the page and describe different things at will. They can move forwards or backwards through text, read text at the character level (for example to confirm spelling), and pause and restart later, among other useful operations. 
+
+Many HTML element nodes can contain attributes. A link may use a `title` attribute, like this ([though I do not recommend it](https://github.com/springernature/frontend-playbook/blob/main/accessibility/common-remediations/html-attributes.md)): `<a href="/" title="don't do this">my cool website address</a>`. The alt attribute is also of this type: `<img alt="my cool alt text" src="...">`. Attributes can't be interacted with in the same way as the rest of the page - you can't move forwards or backwards through the text, you can't read the text at the character level, and you can't pause the screen reader and restart later where you left off. 
```

Comment:
might (though I do not recommend it) read better after the example, for me it was a bit difficult to read.

Reply:
I was wondering why we are promoting an a11y anti-pattern "even though we don't recommend it", but I suppose if its a widespread practice then it does good to cast it more into disrepute.

Reply:
@jpw 2 reasons: 

1. There just aren't that many attributes to choose from that are exposed as announcements to screen readers. You're not going to get a data attribute exposed, for example, or an href, or size information etc, so not much point invoking them here as examples. I really only have a couple of things I can choose from - `title`, which I did use, or `aria-label` / `aria-description`, both of which would suffer similar problems and need to be even more heavily caveated, and that we don't currently have Playbook guidance written up on (it's on my todo list). If you can think of another possibility I'd be very glad to hear it, though.
2. You're correct that `title` abuse is a VERY widespread practice, and I take every opportunity I get to say it shouldn't be done 😅 



@emsel5201 How do you feel about this as a rewrite? 

```suggestion
Many HTML element nodes can contain attributes. [A link could (but shouldn't) use a `title` attribute](https://github.com/springernature/frontend-playbook/blob/main/accessibility/common-remediations/html-attributes.md)), like this: `<a href="/" title="don't do this">my cool website address</a>`. The alt attribute is also of this type: `<img alt="my cool alt text" src="...">`. Attributes can't be interacted with in the same way as the rest of the page - you can't move forwards or backwards through the text, you can't read the text at the character level, and you can't pause the screen reader and restart later where you left off. 
```

Reply:
Yes, I like the rewrite!